test/test_images/envvars/envvars.go

```diff
@@ -26,17 +27,43 @@ import (
 	"github.com/knative/serving/test"
 )
 
-func handler(w http.ResponseWriter, r *http.Request) {
-	log.Print("Env vars test app received a request.")
-	svc := os.Getenv("K_SERVICE")
-	cfg := os.Getenv("K_CONFIGURATION")
-	rev := os.Getenv("K_REVISION")
-	fmt.Fprintf(w, "Here are our env vars service:%s - configuration:%s - revision:%s", svc, cfg, rev)
+//Environment variables that "SHOULD" be set by knative implementation
+var shouldSet = []string{ "K_SERVICE", "K_CONFIGURATION", "K_REVISION"}
```

Comment:
shouldSet and mustSet are defined in both test_image and tests, can they somehow share the code?

Reply:
I did think of that, but I saw that OpenContainers use a different model where it looks like they do code-gen from spec. I was not sure if we would be moving to that model. Just as a layer of safety I have ensure that they match both in numbers and key-names, but I understand your point, will wait for Evan to comment to see whats the plan on having a single source of truth for specifying the expected keys.

Reply:
I don't think we have a plan yet for auto-generating conformance tests based on the spec. If you have a specific idea or proposal, I'd be interested, but I don't have the bandwidth to work on that in the next ~year.

Reply:
Got it. For this change taking into consideration your comment on a more permissive parsing model I have moved environment variables set to conformance/runtime_contact_types.go I will take a deeper look at how things are done in open-container and see if we can apply that or come up with a different approach as an action item.